### ISSUE TITLE
Consider adding "does" to SeleneElement

### ISSUE THREAD

**Issue by kylefix** [yashaka/selene#167]
You could add this:

```
def does(self: SeleneElement, condition) -> bool:
    return is_matched(condition, self)

SeleneElement.does = does
```

In order to be able to do things like this:

```
s(HomeLocators.CURRENT_LANGUAGE).does(
            have.exact_text("FR"))
```

**Comment by SergeyPirogov** [comment]
@kylefix Hi, I don't see any real profit for this. We have should methods and all checks and assertions should be performed like s("locator").should(have.text("hello"))

**Comment by ibalagurov** [comment]
@SergeyPirogov sometimes you don't want to wait element. just to know is it exist or visible. Example: if you want to implement visit page by direct link if it don't open and you use one browser session for better performance so you don't know state of previous test.

**Comment by kylefix** [comment]
Yes, and should throws an error if false right? This is just a boolean.

**Comment by SergeyPirogov** [comment]
I think we can add .has(), so that it will look like

```
if(s("locator").has(text("")))
````

**Comment by aleksandr-kotlyar** [comment]
Closing as already covered in Selene 2.x.

The requested boolean check API is available as:

```python
element.matching(condition)
````

for an immediate non-waiting check, and:

```python
element.wait_until(condition)
```

for a wait-based boolean check.

So we do not plan to add `.does(...)` / `.has(...)` aliases to `SeleneElement`.
They would increase API surface without adding new behavior.
